File: Leetcode/Daily/22-05-2024/prob.go
func isPal(s string, start int,end int) bool{
    for start<=end{
            if(s[start]!=s[end]){
                return false
            }
            start++
            end--
        }

        return true
}

func f(ind int,s string,ans *[][]string,v *[]string){
    if ind==len(s) {
            temp := make([]string, len(*v))
            copy(temp, *v)
            *ans = append(*ans, temp)
            return
    }
    for i:=ind;i<len(s);i++{
        if !isPal(s,ind,i) {
            continue
        }
        *v = append(*v,s[ind:i+1])
        f(i+1,s,ans,v)
        *v = (*v)[:len(*v)-1]
    }
}

func partition(s string) [][]string {
    var ans [][]string
    var v []string
    ind := 0

    f(ind,s,&ans,&v)

    return ans
}

File: Leetcode/Daily/22-05-2024/prob.cpp
class Solution {
public:
    
    void f(int ind, vector<vector<string>>&ans,vector<string>&path,string& s){
        if(ind==s.size()){
            ans.push_back(path);
            return;
        }

        for(int i=ind;i<s.size();i++){
            if(!isPal(s,ind,i))continue;
            path.push_back(s.substr(ind,i-ind+1));
            f(i+1,ans,path,s);
            path.pop_back();
        }
    }

    bool isPal(string s,int start,int end){
        while(start<=end){
            if(s[start]!=s[end]){
                return false;
            }
            start++;
            end--;
        }

        return true;
    }
    
    
    vector<vector<string>> partition(string s) {
       vector<vector<string>>ans;
       vector<string>path;
       f(0,ans,path,s);

       return ans;     
    }
};